src/index_runner/es_indexers/annotated_metagenome_assembly.py

```diff
@@ -136,20 +137,21 @@ def index_annotated_metagenome_assembly(obj_data, ws_info, obj_data_v1):
     workspace_id = info[6]
     object_id = info[0]
     version = info[4]
-
     ama_id = f"{_NAMESPACE}::{workspace_id}:{object_id}"
     ver_ama_id = f"{_VER_NAMESPACE}::{workspace_id}:{object_id}:{version}"
 
     if not data.get('features_handle_ref'):
         raise Exception("AnnotatedMetagenomeAssembly object does not have features_handle_ref"
                         " field. Can not index features to ElasticSearch.")
 
-    # Download features file
-    features_handle_ref = data.get('features_handle_ref')
-    features_file_gz_path = _DIR + "/features.json.gz"
-    handle_id_to_file(features_handle_ref, features_file_gz_path)
-
-    for doc in _index_ama(features_file_gz_path, data, ama_id, ver_ama_id):
-        yield doc
-    # remove zipped file
-    os.remove(features_file_gz_path)
+    try:
```

Comment:
you can move the assignment to features_handle_ref and features_file_gz_path outside of the try, above it.